src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,30 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as json decoded.
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    public function getJsonDecoded(int $depth = 512, int $options = 0)
```

Comment:
I find the `getJsonDecoded()` confusing ... is it decoding the entire Request as JSON? Is it returning only its body/content? Only the headers? etc.

Proposals:

* `getJsonContentAsArray()`
* `getJsonBodyAsArray()`
* `getParsedJsonContent()`
* `getParsedJsonBody()`
* `getContentAsParsedJson()`
* `getBodyAsParsedJson()`
* etc.

I know the proposed names are longer ... but people usually read more code than they write (and most people use autocompleting IDEs anyway) so it's better to optimize for reading code instead of saving a few keystrokes when writing it.

Reply:
Or `asArray()` as I proposed before. But I'm not sure we should add the method personally. I understand the objections that have been raised.